src/bin/clean.rs

```diff
@@ -0,0 +1,31 @@
+use std::io;
+use std::io::Write;
+use std::fs;
+use clap::{App, ArgMatches, SubCommand};
+use mdbook::MDBook;
+use mdbook::errors::Result;
+use get_book_dir;
+
+use std::error::Error;
+
+// Create clap subcommand arguments
+pub fn make_subcommand<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("clean")
+        .about("Delete built book")
+}
+
+// Init command implementation
+pub fn execute(args: &ArgMatches) -> Result<()> {
+    let mut book_dir = get_book_dir(args);
+    book_dir.push("book");
+
+    match fs::remove_dir_all(book_dir.to_str().unwrap()) {
+        Ok(_) => (),
+        m @ _ => {
+            error!("Can't delete built book");
+            return Err("error message")
```

Comment:
You aren't returning a `Result<(), Error>` here, it's just a string. If you change it to `Err(Error::from("error message"))` it'll compile.

Reply:
@Michael-F-Bryan, i rewrote as:
```
pub fn execute(args: &ArgMatches) -> Result<()> {
    let mut book_dir = get_book_dir(args);
    book_dir.push("book");

    match fs::remove_dir_all(book_dir.to_str().unwrap()) {
        Ok(_) => Ok(()),
        Err(_) => {
            error!("Can't delete built book");
            return Err(Error::from("Can't delete built book"))
        },
    }
}
```
Still it gives error: